explainer.md

```diff
@@ -0,0 +1,255 @@
+* Entry/Exit Transitions : A UI element animates as it exits or enters the screen. This [issue](https://github.com/WICG/shared-element-transitions/issues/37) shows an example.
+
+These transitions should be feasible in SPAs (Single Page Apps) and MPAs (Multi Page Apps).
+
+# Design
+Let's take the example below which shows how the API can be used by a developer to animate the background and a shared element on a same origin navigation (MPA).
+
+### Old Document
+```
+<html>
+<head>
+  <style>
+    body {
+      background-color: blue;
+    }
+    .animated {
+      width: 100px;
+      height: 100px;
+      background-color: red;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header" id="header"></div>
+</body>
+</html>
+```
+
+### New Document
+```
+<html>
+<head>
+  <style>
+    body {
+      background-color: lightblue;
+    }
+    .animated {
+      position: relative;
+      top: 100px;
+      width: 200px;
+      height: 200px;
+      background-color: red;
+    }
+    
+    @keyframes fadeIn {
+      0% {opacity: 0;}
+      100% {opacity: 1;}
+    }
+    
+    @keyframes fadeOut {
+      0% {opacity: 1;}
+      100% {opacity: 0;}
+    }
+    
+    ::shared-new(#root), ::shared-new(#shared-header) {
+      animation: fadeIn 1s;
+    }
+
+    ::shared-old(#root), ::shared-old(#shared-header) {
+      animation: fadeOut 1s;
+    }
+    
+    ::shared-container(#shared-header) {
+      transition: all 1s ease-in;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header" id="header"></div>
+</body>
+</html>
+```
+
+The steps taken by the browser during the transition are as follows.
+
+1. When a navigation is initiated on the old Document, create the following pseudo elements in the top layer :
+    a. A container and child replaced element for each element with shared-id attribute. These are identified via ::shared-container(#shared-id) and ::shared-old(#shared-id) respectively.
+    b. A replaced element for the root/html element identified via ::shared-old(#root).
+
+2. Apply the following UA stylesheet to the pseudo elements on the old page :
+```
+::shared-old(#root), ::shared-container(#shared-header) {
+  position: fixed;
+  top: 0px;
+  left: 0px;
+}
+
+::shared-container(#shared-header) {
+  // Container sized to the element's border-box size.
+  width: 100px;
+  height: 100px;
+  
+  // A transform mapping the element to its quad in viewport space.
+  transform: translate(0px, 0px);
+}
+
+::shared-old(#shared-header) {
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  content: element(#header);
+}
+
+::shared-old(#root) {
+  width: 100vw;
+  height: 100vh;
+  // The output of element() function on the root element.
+  content: element(html);
+}
+```
```

Comment:
#shared-header {
  opacity: 0
}

Reply:
Alternative "element() doesn't recurse into shared element descendants"

Reply:
Another reason I was avoiding saying we're overriding the opacity to 0 on shared elements is because this will be visible to script which can't be done in cross-origin.